### ISSUE TITLE
Question: Sanic blocking on time.sleep in async function

### ISSUE THREAD

**Issue by rubenvdham** [channelcat/sanic#1184]
I'm currently writing an authentication API (using sanic-auth). This is my Login route+function:

`@app.route('/login', methods=['GET', 'POST'])
async def login(request):
    if request.method == "GET":
        return response.html(env.get_template("login.html").render())
    if request.method == 'POST':
        validation = await validate_login_data(request)
        if validation[0] and validation[1].check_password(request.form.get("password")):
            auth.login_user(request,validation[1])
            return response.redirect("/")
        else:
            raise loginException("Password incorrect")
    return response.text("Notloggedin")`

The function `async def validate_login_data(request)` (which is awaited) contains one `time.sleep(5) ` for testing purposes. The time.sleep simulates a super long database query/blocking event. When I try to login with 2 different browserwindows to simulate multiple users logging in at the same time, the first login attempt blocks Sanic in handling requests. 
I expected this would not happen because the `validate_login_data(request)` function is being awaited and in the 5 seconds of waiting/querying (technically) other requests could be served. What am I doing wrong? I guess I'm not getting the async/await idea behind Python or maybe the time.sleep stops the whole event loop thingy, although the function is still async.
I would really like to find out what is actually going on:)
Thanks in advance!

**Comment by argaen** [comment]
You have to use `asyncio.sleep` instead of `time.sleep`

**Comment by rubenvdham** [comment]
Thanks! However, suppose user 1 logins in and my database query takes 3 seconds. Then, a second user tries to retrieve the loginpage. Does it have to wait for the database query of user 1 to finish to retrieve the login page? Or is the response sent when the database query is still finishing so user 2 does not experience any lag loading the login page?

**Comment by argaen** [comment]
No it won't wait if all the calls you do are non blocking. `time.sleep` is blocking and it doesn't play nice with asyncio, that's why you have to use `asyncio.sleep`.

**Comment by scottwernervt** [comment]
> Does it have to wait for the database query of user 1 to finish to retrieve the login page?

No, as you should be using an async compatible database driver like [asyncpg](https://github.com/MagicStack/asyncpg) with a connection pool. Using an async compatible driver will make the database query non-blocking.
